jobs/azure-discovery/src/main/java/com/tmobile/pacbot/azure/inventory/file/AssetFileGenerator.java

```diff
@@ -779,15 +841,25 @@ public void generateFiles(List<SubscriptionVH> subscriptions, String filePath) {
 					return;
 				}
 				try {
+					targetTypeList.add("kubernetes");
 					FileManager.generateKubernetesClusterDetailsInfoFile(kubernetesServicesCollector.fetchKubernetesClusterDetails(subscription));
 					log.info("subscription data saved!");
+					targetTypeList.remove("kubernetes");
 				} catch (Exception e) {
 					e.printStackTrace();
 					Util.eCount.getAndIncrement();
 				}
 			});
 
 			executor.shutdown();
+			try {
+				while (!executor.awaitTermination(1, TimeUnit.HOURS)) {
```

Comment:
how you determine that 1 hour is efficient time to wait?

Reply:
I think we are just writing to log if takes more than 1hr, to help us to debug the hanging threads. Correct me @raviteja-mandala

Reply:
yes @santhosh-challa , we are not timing out of the process after 1 hour. We are just logging the collectors which took more that one hour.

Reply:
got it